### ISSUE TITLE
Create Bot Account for Performing Automated Changes

### ISSUE THREAD

**Issue by runewake2** [GoogleContainerTools/kpt#2264]
A number of kpt code changes are performed that would benefit from being automated.

* `make generate` which updates the kpt CLI docs to align with the site state. #1363
* The kpt.dev installation page which must be manually updated after each release so that the new version may be installed.
* Automatically update the kpt dependencies like kustomize. #925

We should create an account or process for making automated changes against the kpt repo to streamline these changes.

There are a number of ways to approach this including:
* GitHub Token - provided by GitHub Actions, can perform actions against the kpt repo only.
* Personal Access Token - provided by a team member, can be scoped to their projects. I do not think this is ideal.
* GitHub App/Bot - Would provide more configurable settings and access restrictions but may be more complex to implement.

We'd need to ensure that whichever path is chosen, it is compatible with the CLA or the changes may be blocked.

**Comment by mortent** [comment]
We already have the "kpt-robot" bot account which used by the old cloud-build release pipeline. I don't think we need separate bot accounts if we do this through github actions (like the new release pipeline).

**Comment by runewake2** [comment]
Do we have the configuration of that account documented anywhere? I couldn't find any commits from the account when I was looking. https://github.com/GoogleContainerTools/kpt/commits?author=kpt-robot

We might be able to re-use it for this.

**Comment by mortent** [comment]
It is used by goreleaser through the cloud build job: https://github.com/GoogleContainerTools/kpt/blob/b8a3eed607432eed570e238054e4d73c04cda95f/release/tag/cloudbuild.yaml#L68
You can see that releases that are done with the cloud build job shows up as being created by the kpt-robot account: https://github.com/GoogleContainerTools/kpt/releases/tag/v0.39.3

**Comment by runewake2** [comment]
We need an email to enable in our CLA and attach to commits. I couldn't find one in that config when I checked or attached to the `kpt-robot` account.

Without this, any PR opened by the robot will be blocked by our SignCLA check.

**Comment by ciaranjohnston** [comment]
No identifiable need, create a new issue if it arises again
